trulens_eval/trulens_eval/streamlit.py

```diff
@@ -0,0 +1,274 @@
+import asyncio
+import json
+import math
+from typing import List
+
+# https://github.com/jerryjliu/llama_index/issues/7244:
+asyncio.set_event_loop(asyncio.new_event_loop())
+
+from millify import millify
+from pydantic import BaseModel
+import streamlit as st
+from streamlit_pills import pills
+
+from trulens_eval import Tru
+from trulens_eval.database.legacy.migration import MIGRATION_UNKNOWN_STR
+from trulens_eval.react_components.record_viewer import record_viewer
+from trulens_eval.schema.feedback import FeedbackCall
+from trulens_eval.schema.feedback import FeedbackDefinition
+from trulens_eval.schema.record import Record
+from trulens_eval.utils import display
+from trulens_eval.utils.python import Future
+from trulens_eval.ux import styles
+from trulens_eval.ux.components import draw_metadata
+from trulens_eval.ux.styles import CATEGORY
+
+tru = Tru()
+
+
+class FeedbackDisplay(BaseModel):
+    score: float = 0
+    calls: List[FeedbackCall]
+    icon: str
+
+
+def trulens_leaderboard(app_ids: List = None):
+    """
+    Render the leaderboard page.
+
+    Args:
+
+        app_ids : A list of application IDs (default is None)
+
+    !!! example
+
+        ```python
+        from trulens_eval import streamlit as trulens_st
+
+        trulens_st.trulens_leaderboard()
+        ```
+    """
+    tru = Tru()
+
+    lms = tru.db
+    df, feedback_col_names = lms.get_records_and_feedback([])
+    feedback_defs = lms.get_feedback_defs()
+    feedback_directions = {
+        (
+            row.feedback_json.get("supplied_name", "") or
+            row.feedback_json["implementation"]["name"]
+        ): row.feedback_json.get("higher_is_better", True)
+        for _, row in feedback_defs.iterrows()
+    }
+
+    if df.empty:
+        st.write("No records yet...")
+        return
+
+    df = df.sort_values(by="app_id")
+
+    if df.empty:
+        st.write("No records yet...")
+
+    apps = list(df.app_id.unique()) if app_ids is None else app_ids
```

Comment:
probably should just replace this with:
```
if app_ids is None:
    app_ids = list(df.app_id.unique())
```
and then use `app_ids` instead of `apps` since you are dealing with app ids and not apps. It won't change the variable for the caller of this function don't worry.